src/common/store/handshake.test.ts

```diff
@@ -0,0 +1,130 @@
+
+import slice, {
+  createKeyPair,
+  createKeyPairWorker,
+  postHandshake,
+  putJson,
+  sendPublicKey,
+  setHandshaken,
+  setKeyPair,
+  setPublicKeyWorker,
+  setTargetPublicKey
+} from './handshake';
+
+describe('Handshake', () => {
+  describe('setHandshaken', () => {
+    it('sets isHandshaken to the payload', () => {
+      expect(slice.reducer({ isHandshaken: false }, setHandshaken(true))).toStrictEqual({
+        isHandshaken: true
+      });
+
+      expect(slice.reducer({ isHandshaken: true }, setHandshaken(false))).toStrictEqual({
+        isHandshaken: false
+      });
+    });
+  });
+
+  describe('setKeyPair', () => {
+    it('sets publicKey and privateKey to the payload', async () => {
+      const keyPair = await createHandshakeKeyPair();
+      expect(slice.reducer({ isHandshaken: false }, setKeyPair(keyPair))).toStrictEqual(
+        expect.objectContaining({
+          ...keyPair
+        })
+      );
+    });
+  });
+
+  describe('setTargetPublicKey', () => {
+    it('sets targetPublicKey to the payload', () => {
+      expect(
+        slice.reducer({ isHandshaken: false }, setTargetPublicKey(fEncryptionPublicKey))
+      ).toStrictEqual(
+        expect.objectContaining({
+          targetPublicKey: fEncryptionPublicKey
+        })
+      );
+    });
+  });
+});
+
+describe('putJson', () => {
+  it('puts an action if allowInsecure is set or if the action is handshake/sendPublicKey', async () => {
+    const action = JSON.stringify(sendPublicKey(fEncryptionPublicKey));
+    await expectSaga(putJson, action)
+      .put({ ...sendPublicKey(fEncryptionPublicKey), remote: true })
+      .silentRun();
+
+    const insecureAction = JSON.stringify(setHandshaken(true));
+    await expectSaga(putJson, insecureAction, true)
+      .put({ ...setHandshaken(true), remote: true })
+      .silentRun();
+  });
+
+  it('decrypts an encrypted action', async () => {
+    const insecureAction = JSON.stringify(setHandshaken(true));
+    const encryptedAction = encrypt(
+      fEncryptionPublicKey,
+      Buffer.from(insecureAction, 'utf-8')
+    ).toString('hex');
+    await expectSaga(putJson, JSON.stringify({ data: encryptedAction }), true)
+      .withState({ handshake: { isHandshaken: true, privateKey: fEncryptionPrivateKey } })
+      .put({ ...setHandshaken(true), remote: true })
+      .silentRun();
+  });
+
+  it('does nothing on invalid JSON', async () => {
+    await expectSaga(putJson, 'foo bar').silentRun();
+  });
+});
+
+describe('createKeyPairWorker', () => {
+  it('generates a keypair and dispatches setKeyPair', async () => {
+    const keyPair = await createHandshakeKeyPair();
+
+    await expectSaga(createKeyPairWorker, createKeyPair()).put(setKeyPair(keyPair)).silentRun();
+  });
+
+  it('sends the public key if the payload is true', async () => {
+    const keyPair = await createHandshakeKeyPair();
+
+    await expectSaga(createKeyPairWorker, createKeyPair(true))
```

Comment:
Can we document this somewhere in the code that the renderer initializes the handshake process with this call?